### DISCUSSION TITLE
Heavy memory usage and slow rendering when zooming in long audios

### DISCUSSION THREAD

**Discussion by larsrgit** [katspaugh/wavesurfer.js#3193] [Q&A]
I have a long audio file (~1hour), the raw pcm data is around 200Mb. Zooming in (minPxPerSec=300)  increases Memory usage by up to 4GB compared to staying zoomed out (minPxPerSec=50).
Especially when scrolling to a part at the end of the audio, the rendering takes up to a minute when zoomed in.

Is this expected behaviour? I guess when the renderer renders the whole 1hour of audio 3600s x 300PxPerSec x 100px height the whole rendering is 108 million pixel in size. But even that should not be 4GB.

**Comment by katspaugh** [comment]
That's not good. All those pixels shouldn't be held in the memory anyway. There might be a leak somewhere.

**Comment by jenga555** [comment]
I've also found that the timeline plugin reports the wrong numbers if you zoom in like 10x and then try to go back. It works fine zooming in, but zooming back out seems to be the problem. (and this was on only a 4 min audio file)

**Reply by katspaugh** [reply]
Thanks! Created an issue for that: https://github.com/katspaugh/wavesurfer.js/issues/3222
